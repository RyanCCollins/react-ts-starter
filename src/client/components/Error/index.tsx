import * as React from 'react';
import { Box, Paragraph } from 'components';

export interface ErrorProps {
  message: JSX.Element | string;
  onClick?: React.EventHandler<any>;
}

export default function Error({
  message,
  onClick,
}: ErrorProps): JSX.Element {
  return (
    <Box
      backgroundColor="#ff324d"
      boxSize={{ horizontal: 'medium' }}
      pad="small"
      onClick={onClick}
      alignItems="center"
      selectable
    >
      <Paragraph paragraphSize="large" color="white">{message}</Paragraph>
    </Box>
  );
};
