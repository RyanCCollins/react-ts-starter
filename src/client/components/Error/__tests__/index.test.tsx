import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Error from '../';
import { ErrorPropsRequired, ErrorPropsOther } from './__mocks__/errorMocks.mock';

describe('<Error /> default', () => {
  it('should render an Error element with only the required props', () => {
    const wrapper = shallow(
      <Error {...ErrorPropsRequired} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render an Error element with other props', () => {
    const wrapper = shallow(
      <Error {...ErrorPropsOther} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('The onClick prop should trigger with an onClick event', () => {
  let mockFunction;
  let error;

  beforeEach(() => {
    mockFunction = jest.fn();
    error = mount(
      <Error {...ErrorPropsRequired} onClick={mockFunction} />,
    );
  });

  it('Error requires onClick prop', () => {
    expect(error.props().onClick).toBeDefined();
  });

  it('Error rendered with Box - an HTML div element', () => {
    const renderedError = error.find('div').first();
    expect(renderedError).toBeDefined();
  });

  it('Error click calls mockFunction', () => {
    const renderedError = error.find('div').first();
    renderedError.simulate('click');
    expect(mockFunction).toBeCalled();
  });
});
