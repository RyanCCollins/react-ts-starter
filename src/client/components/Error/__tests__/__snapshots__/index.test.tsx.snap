// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> default should render an Error element with only the required props 1`] = `
<Box
  alignItems="center"
  backgroundColor="#ff324d"
  boxSize={
    Object {
      "horizontal": "medium",
    }
  }
  pad="small"
  selectable={true}
>
  <styled.p
    color="white"
    paragraphSize="large"
  >
    Mock error message
  </styled.p>
</Box>
`;

exports[`<Error /> default should render an Error element with other props 1`] = `
<Box
  alignItems="center"
  backgroundColor="#ff324d"
  boxSize={
    Object {
      "horizontal": "medium",
    }
  }
  pad="small"
  selectable={true}
>
  <styled.p
    color="white"
    paragraphSize="large"
  >
    Mock error message
  </styled.p>
</Box>
`;
